src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.c

```diff
@@ -21,39 +44,477 @@ const ActorInit En_Ex_Item_InitVars = {
+    }
+    if (this->timer != 0) {
+        if (this->prizeRotateTimer != 0) {
+            sp3C = 250.0f;
+            if (this->type == EXITEM_BOMBCHUS_BOWLING) {
+                sp3C = 220.0f;
+            }
+            tmpf1 = globalCtx->view.lookAt.x - globalCtx->view.eye.x;
+            tmpf2 = globalCtx->view.lookAt.y - globalCtx->view.eye.y;
+            tmpf3 = globalCtx->view.lookAt.z + sp3C - globalCtx->view.eye.z;
+            tmpf4 = sqrtf(SQ(tmpf1) + SQ(tmpf2) + SQ(tmpf3));
+
+            tmpf5 = (tmpf1 / tmpf4) * 5.0f;
+            tmpf6 = (tmpf2 / tmpf4) * 5.0f;
+            tmpf7 = (tmpf3 / tmpf4) * 5.0f;
+
+            tmpf1 = globalCtx->view.eye.x + tmpf5 - this->actor.posRot.pos.x;
+            tmpf2 = globalCtx->view.eye.y + tmpf6 - this->actor.posRot.pos.y;
+            tmpf3 = globalCtx->view.eye.z + tmpf7 - this->actor.posRot.pos.z;
+
+            this->actor.posRot.pos.x += (tmpf1 / tmpf4) * 5.0f;
+            this->actor.posRot.pos.y += (tmpf2 / tmpf4) * 5.0f;
+            this->actor.posRot.pos.z += (tmpf3 / tmpf4) * 5.0f;
+        }
+    } else {
+        // parent
+        osSyncPrintf(VT_FGCOL(GREEN) " ☆☆☆☆☆ 母親ー？     ☆☆☆☆☆ %x\n" VT_RST, this->actor.parent);
+        // Can it move?
+        osSyncPrintf(VT_FGCOL(GREEN) " ☆☆☆☆☆ 動いてねー？ ☆☆☆☆☆ %x\n" VT_RST, this->actor.parent->update);
+        if ((this->actor.parent != NULL) && (this->actor.parent->update != NULL)) {
+            ((EnBomBowlPit*)this->actor.parent)->exItemDone = 1;
+            // It can't move!
+            osSyncPrintf(VT_FGCOL(GREEN) " ☆☆☆☆☆ さぁきえるぞ！ ☆☆☆☆☆ \n" VT_RST);
+        }
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnExItem_SetupBowlCounter(EnExItem* this, GlobalContext* globalCtx) {
+    this->actor.posRot.rot.y = this->actor.shape.rot.y = 0x4268;
+    this->actionFunc = EnExItem_BowlCounter;
+}
+
+void EnExItem_BowlCounter(EnExItem* this, GlobalContext* globalCtx) {
+    this->actor.shape.rot.y += 0x800;
+    if (this->killItem) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnExItem_ExitChest(EnExItem* this, GlobalContext* globalCtx) {
+    this->actor.shape.rot.y += 0x1000;
+    if (this->timer != 0) {
+        if (this->timer == 1) {
+            this->chestKillTimer = 20;
+        }
+    } else {
+        this->actor.velocity.y = 0.0f;
+        if (this->chestKillTimer == 0) {
+            Actor_Kill(&this->actor);
+        }
+    }
+    Actor_MoveForward(&this->actor);
+}
+
+void EnExItem_FairyMagic(EnExItem* this, GlobalContext* globalCtx) {
+    this->actor.shape.rot.y += 0x800;
+}
+
+void EnExItem_TargetPrizeApproach(EnExItem* this, GlobalContext* globalCtx) {
+    f32 tmpf1;
+    f32 tmpf2;
+    f32 tmpf3;
+    f32 tmpf4;
+    f32 tmpf5;
+    f32 tmpf6;
+    f32 tmpf7;
+
+    Math_ApproachF(&this->scale, 0.8f, 0.1f, 0.02f);
+    if (!this->stopRotate) {
+        this->actor.shape.rot.y += 0x1000;
+        if ((this->prizeRotateTimer == 0) && ((this->actor.shape.rot.y & 0xFFFF) == 0x9000)) {
```

Comment:
Same suggestion here
```suggestion
        if ((this->prizeRotateTimer == 0) && (this->actor.shape.rot.y == 0x9000)) {
```